docs/docs/references/embedding.mdx

```diff
@@ -69,32 +66,35 @@ Clicking `generate & copy url` will generate an embed URL based on the configura
 
 You can embed this one-off URL directly into your application, but you will manually need to update the URL each time the embed URL expires. Alternatively, you can add the [`code snippet`](#code-snippet) to your app to automatically generate embed URLs in your application.  
 
-### Code snippet
+## Code snippet
 
 Although you can generate URLs directly from Lightdash with a long expiration, it is recommended to generate your own JWT embed tokens in your backend with a short expiration using your `secret` to make sure people can't be using embed URLs outside your app. 
 
 To make this easier to integrate, we included some code snippets you can copy and use in your app to generate a valid embed URL.
 
-### Passing external variables to user attributes
+<img src={EmbedDeveloper} width="668" height="466" style={{display: "block", margin: "0 auto 20px auto"}}/>
 
-For embedded content, you can either set user attributes to constant values, or you can adjust the value of the user attribute based on a variable coming from your app. 
+### User metadata
 
-For example, if I want users in my app to only see data filtered to their company, I can use a variable user attribute. 
+You can pass user metadata (specifically, an external user ID) from your application so that anytime someone views your embedded dashboard and runs queries in Lightdash, these query logs are enriched with this user metadata. Specifcally, this user metadata gets added to the [query tags](/references/usage-analytics.mdx#query-tags) for queries run in Lightdash.
 
-To do this, I would create a user attribute in Lightdash called `company_name` and set some default (e.g. `null`) then, I would pass the Lightdash an external value for company name in the embed code snippet, like:
+To assign an external ID, you can update the embed code snippet like this:
 
 ```node
 import jwt from 'jsonwebtoken';
 const LIGHTDASH_EMBED_SECRET = 'secret'; // replace with your secret
-const projectUuid = '21eef0b9-5bae-40f3-851e-9554588e71a6';
-const userCompanyName = await getUserCompanyNameFromDatabase();
+const projectUuid = 'my-project-uuid';
 const data = {
+    user: {
+      externalId: 'your_user_id_123', // Add this to assign an external ID
+    },
     content: {
-        type: 'dashboard',
-        dashboardUuid: 'a392ed6d-4c53-4102-89b7-1c57a87df391',
+      type: 'dashboard',
+      dashboardUuid: 'your dashboard uuid'
     },
-    userAttributes: {"company_name": userCompanyName},
 };
 const token = jwt.sign(data, LIGHTDASH_EMBED_SECRET, { expiresIn: '1 hour' });
 const url = `https://analytics.lightdash.cloud/embed/${projectUuid}/#${token}`;
-```
\ No newline at end of file
+```
+
+If you don't assign an external ID in your embed code, then we will automatically generate and an assign an external ID based on the embed token.
```

Comment:
yes, this is correct, more specifically, by default we use the `iat` (issued at) from the token, and it will be unique for a generated token.

Reply:
great, thanks!